### PR TITLE
Fix setting metadata on aws that contains special cars

### PR DESCRIPTION
**Pull request by cbookg**
- Quotes and backslashes need to be escaped
- Also make the json we write more compact as there is a size limit

### CODE REVIEW

File: startupscript/butane/005-parse-devcontainer.sh [hunk 1]
```diff
@@ -38,6 +38,6 @@ export PATH="/opt/bin:$PATH"
 # shellcheck source=/dev/null
 source /home/core/metadata-utils.sh
 readonly JSONC_STRIP_COMMENTS=/home/core/jsoncStripComments.mjs
-DEVCONTAINER_CUSTOMIZATIONS=$(${JSONC_STRIP_COMMENTS} < "${DEVCONTAINER_CONFIG_PATH}" | jq .customizations.workbench)
+DEVCONTAINER_CUSTOMIZATIONS=$(${JSONC_STRIP_COMMENTS} < "${DEVCONTAINER_CONFIG_PATH}" | jq -c .customizations.workbench)
```
Comment (mbookman) [inline startupscript/butane/005-parse-devcontainer.sh:41]:
```suggestion
DEVCONTAINER_CUSTOMIZATIONS="$("${JSONC_STRIP_COMMENTS}" < "${DEVCONTAINER_CONFIG_PATH}" | jq -c .customizations.workbench)"
```

File: startupscript/butane/aws/metadata-utils.sh [hunk 2]
```diff
@@ -49,17 +49,18 @@ readonly -f get_guest_attribute
 function set_metadata() {
   local key="${1}"
   local value="${2}"
-  
+  local escaped="${value//[\"\\]/\\&}"
```
Comment (mbookman) [inline startupscript/butane/aws/metadata-utils.sh:52]: Please add a comment indicating the substitution that this command is doing and why (point to AWS documentation if it exists).

**Review by mbookman — CHANGES_REQUESTED** [review]

**Review by yuhuyoyo — APPROVED** [review]

File: startupscript/butane/aws/metadata-utils.sh (hunk 2 above)
Comment (cbookg) [reply startupscript/butane/aws/metadata-utils.sh:55]: done

**Review by mbookman — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**